src/material-experimental/mdc-tabs/harness/tabs-harness.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessPredicate, TestElement} from '@angular/cdk-experimental/testing';
+import {TabGroupHarnessFilters} from './tabs-harness-filters';
+
+/**
+ * Harness for interacting with a standard mat-tab-group in tests.
+ * @dynamic
+ */
+export class MatTabGroupHarness extends ComponentHarness {
+  static hostSelector = '.mat-tab-group';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a radio-button with
+   * specific attributes.
+   * @param options Options for narrowing the search
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: TabGroupHarnessFilters = {}): HarnessPredicate<MatTabGroupHarness> {
+    return new HarnessPredicate(MatTabGroupHarness);
+  }
+
+  private _tabs = this.locatorForAll(MatTabHarness);
+
+  /** Gets all tabs of the tab group. */
+  async getTabs(): Promise<MatTabHarness[]> {
+    return this._tabs();
+  }
+
+  /** Gets the selected tab of the tab group. */
+  async getSelectedTab(): Promise<MatTabHarness> {
+    for (let tab of await this.getTabs()) {
+      if (await tab.isSelected()) {
```

Comment:
Promise.all

Reply:
where you want to use `Promise.all` here? Also isn't it faster to only check sequentially and exit earlier without awaiting remaining tabs?

Reply:
I think `Promise.all` parallelizes them, so it's better to do that than do them sequentially and find out that its the last one

Reply:
That's a question of probability, but I assume that your proposal has a higher chance being faster. will update :smile:

Reply:
it has a better worst-case

Reply:
true, done! please have another look